packages/fields/src/fields/parent/parent-edit.tsx

```diff
@@ -0,0 +1,351 @@
+/**
+ * WordPress dependencies
+ */
+import { ComboboxControl, ExternalLink } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import {
+	createInterpolateElement,
+	useCallback,
+	useMemo,
+	useState,
+} from '@wordpress/element';
+// @ts-ignore
+import { store as coreStore } from '@wordpress/core-data';
+import type { DataFormControlProps } from '@wordpress/dataviews';
+
+/**
+ * External dependencies
+ */
+import removeAccents from 'remove-accents';
+
+/**
+ * Internal dependencies
+ */
+import { debounce } from '@wordpress/compose';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __, sprintf } from '@wordpress/i18n';
+import type { BasePost } from '../../types';
+import { getTitleWithFallbackName } from './utils';
+import { filterURLForDisplay } from '@wordpress/url';
+
+type TreeBase = {
+	id: number;
+	name: string;
+	[ key: string ]: any;
+};
+
+type TreeWithParent = TreeBase & {
+	parent: number;
+};
+
+type TreeWithoutParent = TreeBase & {
+	parent: null;
+};
+
+type Tree = TreeWithParent | TreeWithoutParent;
+
+function buildTermsTree( flatTerms: Tree[] ) {
+	const flatTermsWithParentAndChildren = flatTerms.map( ( term ) => {
+		return {
+			children: [],
+			...term,
```

Comment:
Interesting how using TypeScript here forced us to declare the Term API and make parent obligatory, making the prop redundant here compared to the [original](https://github.com/WordPress/gutenberg/blob/ce7f1d857b5a35b430c8a37b604d57e2402276da/packages/editor/src/utils/terms.js#L17) that used:

```js
{
  children: [],
  parent: null,
  ...term,
}
```